src/agent/border_agent.cpp

```diff
@@ -285,7 +285,7 @@ static const char *ThreadVersionToString(uint16_t aThreadVersion)
     case 3:
         return "1.2.0";
     default:
-        return "";
+        return NULL;
```

Comment:
assert here?

Reply:
`ThreadVersion` is typically requested from thread devices. Assert here is assuming a thread device will always return thread version of 2 or 3 and we shouldn't make this assumption.